--- ../src_base/minecraft_server/net/minecraft/src/EntityMob.java
+++ ../src/minecraft_server/net/minecraft/src/EntityMob.java
@@ -1,4 +1,6 @@
 package net.minecraft.src;
+
+import soulforge.utils.EventRegistry;
 
 public abstract class EntityMob extends EntityCreature implements IMob
 {
@@ -8,6 +10,7 @@
         this.experienceValue = 5;
     }
 
+    
     /**
      * Called frequently so the entity can update its state every tick as required. For example, zombies and skeletons
      * use this to react to sunlight and start to burn.
