--- ../src_base/minecraft_server/soulforge/utils/GameRegistry.java
+++ ../src/minecraft_server/soulforge/utils/GameRegistry.java
@@ -6,6 +6,7 @@
 import soulforge.itemblock.ItemMultiBlock;
 import net.minecraft.src.Block;
 import net.minecraft.src.CraftingManager;
+import net.minecraft.src.FCAddOnHandler;
 import net.minecraft.src.IChunkProvider;
 import net.minecraft.src.IRecipe;
 import net.minecraft.src.Item;
