--- ../src_base/minecraft/net/minecraft/src/MapGenScatteredFeature.java
+++ ../src/minecraft/net/minecraft/src/MapGenScatteredFeature.java
@@ -5,12 +5,14 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Random;
-import java.util.Map.Entry;
+
+import biomesoplenty.configuration.BOPBiomes;
 
 public class MapGenScatteredFeature extends MapGenStructure
 {
-    private static List biomelist = Arrays.asList(new BiomeGenBase[] {BiomeGenBase.desert, BiomeGenBase.desertHills, BiomeGenBase.jungle, BiomeGenBase.jungleHills, BiomeGenBase.swampland});
+    private static List biomelist = Arrays.asList(new BiomeGenBase[] {BiomeGenBase.desert, BiomeGenBase.desertHills, BiomeGenBase.jungle, BiomeGenBase.jungleHills, BiomeGenBase.swampland, BOPBiomes.timber, BOPBiomes.temperateRainforest});
 
     /** contains possible spawns for scattered features */
     private List scatteredFeatureSpawnList;
