--- ../src_base/minecraft/net/minecraft/src/World.java
+++ ../src/minecraft/net/minecraft/src/World.java
@@ -1,5 +1,6 @@
 package net.minecraft.src;
 
+import java.beans.EventHandler;
 import java.util.ArrayList;
 import java.util.Calendar;
 import java.util.Collection;
@@ -8,6 +9,8 @@
 import java.util.List;
 import java.util.Random;
 import java.util.Set;
+
+import soulforge.utils.EventRegistry;
 
 public abstract class World implements IBlockAccess
 {
@@ -2807,7 +2810,9 @@
                 }
             }
         }
-
+        if (!this.isRemote) {
+        	this.activeChunkSet.addAll(EventRegistry.loadedChunksForDimension(this.provider.dimensionId));
+        }
         this.theProfiler.endSection();
 
         if (this.ambientTickCountdown > 0)
@@ -3302,9 +3307,29 @@
 
         for (int var3 = 0; var3 < this.loadedEntityList.size(); ++var3)
         {
-            Entity var4 = (Entity)this.loadedEntityList.get(var3);
-
-            if ((!(var4 instanceof EntityLiving) || !((EntityLiving)var4).func_104002_bU()) && par1Class.isAssignableFrom(var4.getClass()))
+            Entity entity = (Entity)this.loadedEntityList.get(var3);
+
+            if ((!(entity instanceof EntityLiving) || !((EntityLiving)entity).func_104002_bU() || !((EntityLiving)entity).isFromChunkLoader()) && par1Class.isAssignableFrom(entity.getClass()))
+            {
+                ++var2;
+            }
+        }
+
+        return var2;
+    }
+    
+    /**
+     * Counts how many entities of an entity class exist in the world. Args: entityClass
+     */
+    public int countEntitiesFromChunkLoader(Class par1Class)
+    {
+        int var2 = 0;
+
+        for (int var3 = 0; var3 < this.loadedEntityList.size(); ++var3)
+        {
+            Entity entity = (Entity)this.loadedEntityList.get(var3);
+
+            if ((entity instanceof EntityLiving) && ((EntityLiving)entity).isFromChunkLoader() && par1Class.isAssignableFrom(entity.getClass()))
             {
                 ++var2;
             }
