--- ../src_base/minecraft/net/minecraft/src/WorldServer.java
+++ ../src/minecraft/net/minecraft/src/WorldServer.java
@@ -7,6 +7,8 @@
 import java.util.Random;
 import java.util.Set;
 import java.util.TreeSet;
+
+import soulforge.utils.EventRegistry;
 import net.minecraft.server.MinecraftServer;
 
 public class WorldServer extends World
@@ -267,10 +269,11 @@
         int var1 = 0;
         int var2 = 0;
         Iterator var3 = this.activeChunkSet.iterator();
-
         while (var3.hasNext())
         {
+        	
             ChunkCoordIntPair var4 = (ChunkCoordIntPair)var3.next();
+            
             int var5 = var4.chunkXPos * 16;
             int var6 = var4.chunkZPos * 16;
             this.theProfiler.startSection("getChunk");
@@ -463,7 +466,7 @@
      */
     public void updateEntities()
     {
-        if (this.playerEntities.isEmpty())
+        if (this.playerEntities.isEmpty() && (!isRemote && !EventRegistry.hasLoadedChunksForDimension(provider.dimensionId)))
         {
             if (this.updateEntityTick++ >= 1200)
             {
